# ngsLearn

I document the materials to learn Next Generation Sequencing (NGS) and the learning path.

1. Basic command line and shell script to handle NGS data, i.e. awk, cut, paste, sed, grep.
I will write more example for the useful command.

  An good additional exercise is here. 
  http://www.ark-genomics.org/events-online-training/counting-known-micrornas-five-easy-steps.

2. The customised NGS toolkit, including bedtools, samtools, seqtk, and tabix are then reviewed. These materials are good start point for each of the tools.

  bedtools http://quinlanlab.org/tutorials/bedtools/bedtools.html
	bioawk https://github.com/ialbert/bioawk/blob/master/README.bio.rst
  seqtk https://github.com/lh3/seqtk
  tabix http://www.htslib.org/doc/tabix.html
  samtools http://samtools.github.io/

3. Python for biologist

4. RNA-Seq pipeline, i.e. aligner (bowtie2), splice junction identification (Tophat2), read count, differentially expressed gene analysis and visualization.
5. Exome-Seq pipeline, i.e. aligner (), GATK (), 
6. Chi-Seq pipeline 
Mistakes and erorrs may exists, please do not hestitate to point them out by an email to liuzhebiosci(you know what should be put here)163.com.
